crates/esplora/src/async_ext.rs

```diff
@@ -241,9 +241,9 @@ impl EsploraAsyncExt for esplora_client::AsyncClient {
 
                 let last_index = last_index.expect("Must be set since handles wasn't empty.");
                 let gap_limit_reached = if let Some(i) = last_active_index {
-                    last_index > i.saturating_add(stop_gap as u32)
+                    last_index >= i.saturating_add(stop_gap as u32)
                 } else {
-                    last_index + 1 >= stop_gap as u32
+                    last_index > stop_gap as u32
```

Comment:
In  08f04585e8bd900b9f88708b1193136480cd2061: this shouldn't have changed, `last_index + 1 >= stop_gap` was correct

Reply:
Yes you are absolutely right. I thought that I was simplifying the code but the expressions are not different.
I think that all tests should pass now.
I've changed it back and amended in 95230d7d2b3d662277ec6a476451fd5ff4c3bd08